Update apiDefinition.swagger.json for BoomApp Connector

SagarCIS opened the pull request:
---
### When submitting a connector, please make sure that you follow the requirements below, otherwise your PR might be rejected. We want to make sure you have a well-built connector, a smooth certification experience, and your users are happy :)

If this is your first time submitting to GitHub and you need some help, please sign up for this [session](https://forms.office.com/pages/responsepage.aspx?id=KtIy2vgLW0SOgZbwvQuRaXDXyCl9DkBHq4A2OG7uLpdUMTFJWFFGVUxBNUFZQjZWRUdaOE5BMFkwNS4u).

- [ ] I attest that the connector doesn't already exist on the Power Platform today. I have verified by checking the pull requests in GitHub and searching for the connector on the platform and in the documentation.
- [x] I attest that the connector works and I verified by deploying and testing all the operations.
- [x] I attest that I have added detailed descriptions for all operations and parameters in the swagger file.
- [x] I attest that I have added response schemas to my actions, unless the response schema is dynamic.
- [x] I validated the swagger file, `apiDefinition.swagger.json`, by running the `paconn validate` command.
- [x] As this is a certified connector, I confirm that `apiProperties.json` has a valid brand color and doesn't use an invalid brand color (`#007ee5` or `#ffffff`).

If you are an Independent Publisher, you must also attest to the following to ensure a smooth publishing process (Not applicable for this certified connector):

- [ ] I have named this PR following the pattern "Connector Name (Independent Publisher)".
- [ ] Within this PR markdown file, I have pasted screenshots that show 3 unique operations (actions/triggers) working within a Flow, with screenshots of the flows succeeding.
- [ ] Within this PR markdown file, I have pasted a screenshot from the Test operations section within the Custom Connector UI.
- [ ] If the connector uses OAuth, I have provided detailed steps on how to create an app in the readme.md.

SagarCIS committed:
Update apiDefinition.swagger.json

SagarCIS committed:
Update apiDefinition.swagger.json including the description for existing methods and parameters

BoomerangMessaging committed:
Revise Readme for BoomApp Connect Connector

Updated the description and links for the BoomApp Connect Connector.

BoomerangMessaging committed:
Merge pull request #1 from BoomerangMessaging/feature/readme-update

Revise Readme for BoomApp Connect Connector

andylfcallen committed:
Update apiDefinition.swagger.json

Function updates including push voice replies and Get inbound (add later)

SagarCIS commented:
@microsoft-github-policy-service agree

vmanoharas commented:
Hi @SagarCIS, 

Please fix the swagger validation errors. 

2025-12-23T13:19:46.9524700Z ##[error]The parameter cannot define a schema as it is not in the 'body'. ValuePath: /paths/~1trigger~1v1~1get-inbound/get/parameters/4/schema
2025-12-23T13:19:46.9545041Z ##[error]The 'x-ms-summary' property is required. ValuePath: /paths/~1v1~1get_responses/get/parameters/0
2025-12-23T13:19:46.9546950Z ##[error]The 'x-ms-summary' property is required. ValuePath: /paths/~1v1~1get_responses/get/parameters/1
2025-12-23T13:19:46.9548292Z ##[error]The 'x-ms-summary' property is required. ValuePath: /paths/~1v1~1get_responses/get/parameters/2
2025-12-23T13:19:46.9549377Z ##[error]The 'x-ms-summary' property is required. ValuePath: /paths/~1v1~1get_responses/get/parameters/3
2025-12-23T13:19:46.9550436Z ##[error]The 'x-ms-summary' property is required. ValuePath: /paths/~1v1~1get_responses/get/parameters/4
2025-12-23T13:19:46.9551575Z ##[error]The 'x-ms-summary' property is required. ValuePath: /paths/~1v1~1get_responses/get/parameters/5
2025-12-23T13:19:46.9552826Z ##[error]The 'x-ms-summary' property is required. ValuePath: /paths/~1v1~1get_responses/get/parameters/6
2025-12-23T13:19:46.9553929Z ##[error]The 'x-ms-summary' property is required. ValuePath: /paths/~1v1~1get_responses/get/parameters/7
2025-12-23T13:19:46.9555055Z ##[error]The 'x-ms-summary' property is required. ValuePath: /paths/~1trigger~1v1~1get-inbound/get/parameters/1
2025-12-23T13:19:46.9556871Z ##[error]The 'x-ms-summary' property is required. ValuePath: /paths/~1trigger~1v1~1get-inbound/get/parameters/2
2025-12-23T13:19:46.9558196Z ##[error]The 'x-ms-summary' property is required. ValuePath: /paths/~1trigger~1v1~1get-inbound/get/parameters/3
2025-12-23T13:19:46.9560028Z ##[error]The 'x-ms-summary' property is required. ValuePath: /paths/~1trigger~1v1~1get-inbound/get/parameters/4
2025-12-23T13:19:46.9561123Z ##[error]The 'x-ms-summary' property is required. ValuePath: /paths/~1trigger~1v1~1get-inbound/get/parameters/5
2025-12-23T13:19:46.9562254Z ##[error]The 'x-ms-summary' property is required. ValuePath: /paths/~1trigger~1v1~1get-inbound/get/parameters/6
2025-12-23T13:19:46.9563351Z ##[error]The 'x-ms-summary' property is required. ValuePath: /paths/~1trigger~1v1~1get_all_new_drs/get/parameters/0
2025-12-23T13:19:46.9564415Z ##[error]The 'x-ms-summary' property is required. ValuePath: /paths/~1trigger~1v1~1get_all_new_drs/get/parameters/1
2025-12-23T13:19:46.9566333Z ##[error]The 'x-ms-summary' property is required. ValuePath: /paths/~1trigger~1v1~1get_all_new_drs/get/parameters/2
2025-12-23T13:19:46.9567980Z ##[error]The 'x-ms-summary' property is required. ValuePath: /paths/~1trigger~1v1~1get_all_new_drs/get/parameters/3
2025-12-23T13:19:46.9569502Z ##[error]The 'x-ms-summary' property is required. ValuePath: /paths/~1v1~1get_all_new_drs/get/parameters/0
2025-12-23T13:19:46.9570938Z ##[error]The 'x-ms-summary' property is required. ValuePath: /paths/~1v1~1get_all_new_drs/get/parameters/1
2025-12-23T13:19:46.9572180Z ##[error]The 'x-ms-summary' property is required. ValuePath: /paths/~1v1~1get_all_new_drs/get/parameters/2
2025-12-23T13:19:46.9573741Z ##[error]The 'x-ms-summary' property is required. ValuePath: /paths/~1v1~1get_all_new_drs/get/parameters/3
2025-12-23T13:19:46.9575447Z ##[error]The 'default' response should not have schema definition. Schemas should be defined on expected responses only. ValuePath: /paths/~1v1~1get_responses/get/responses/default
2025-12-23T13:19:46.9578096Z ##[error]The 'default' response should not have schema definition. Schemas should be defined on expected responses only. ValuePath: /paths/~1v1~1MaskTwoWay/post/responses/default
2025-12-23T13:19:46.9579904Z ##[error]The 'default' response should not have schema definition. Schemas should be defined on expected responses only. ValuePath: /paths/~1trigger~1v1~1get-inbound/get/responses/default
2025-12-23T13:19:46.9585442Z ##[error]The 'x-ms-connector-metadata' property is required.

SagarCIS committed:
Merge branch 'microsoft:dev' into feature/boomAppConnectv2

SagarCIS committed:
Fix error schema, x-ms-summary, default, x-ms-connector-metadata

SagarCIS commented:
Hi @vmanoharas ,

Thanks for the feedback. I’ve fixed the swagger validation issues by:
- Removing schema definitions from non-body parameters
- Adding x-ms-summary to all parameters
- Removing schema from default responses

The swagger file has been updated on the same branch and pushed.
Kindly re-review when you get a chance.

Thanks!

vmanoharas commented:
Hi @SagarCIS, 

Thank you for fixing all the swagger validation issues. However, encountered one more error. 

##[error]The connector metadata property 'Categories' is required. ValuePath: /x-ms-connector-metadata

To fix this error, please follow the below steps. 

Correct format:
x-ms-connector-metadata": [
  {
    "propertyName": "Website",
    "propertyValue": "<your-website-url>"
  },
  {
    "propertyName": "Privacy policy",
    "propertyValue": "<your-privacy-policy-url>"
  },
  {
    "propertyName": "Categories",
    "propertyValue": "<your-categories-separated-by-semicolon>"
  }
]
Only two categories are allowed with semicolon ( ; ) separated from this list -  ["AI","Business Management","Business Intelligence","Collaboration","Commerce","Communication","Content and Files","Finance","Data","Human Resources","Internet of Things","IT Operations","Lifestyle and Entertainment","Marketing","Productivity","Sales and CRM","Security","Social Media","Website"]

SagarCIS committed:
Correct format: x-ms-connector-metadata

SagarCIS commented:
Hi @vmanoharas,

Thank you for pointing this out.

I have updated the Swagger definition and added the required x-ms-connector-metadata section, including the mandatory Categories property in the correct format

The swagger file has been updated on the same branch and pushed.
Kindly re-review when you get a chance.

vmanoharas reviewed (approved):
Dear Partner,
 
Congratulations your PR is approved !! We are proud to announce the Brand-New Certification experience for you to certify Power Platform Copilot Connectors. 
 
As next steps:
 
Learn about the new certification experience- [Get your Power Platform connector and plugin certified - Overview](https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Flearn.microsoft.com%2Fen-us%2Fconnectors%2Fcustom-connectors%2Fsubmit-certification&data=05%7C02%7Cv-manoharas%40microsoft.com%7C3fecd63b7d2c4d65d68208dc7bdf57ba%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C638521446390456966%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C0%7C%7C%7C&sdata=uWDvs%2BmTbozw1oPc%2FNDNWJ%2FHgnUSfFlRCB1T8UHBh3A%3D&reserved=0)
Package your connector files- [Prepare Power Platform connector and plugin files for certification](https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Flearn.microsoft.com%2Fen-us%2Fconnectors%2Fcustom-connectors%2Fcertification-submission&data=05%7C02%7Cv-manoharas%40microsoft.com%7C3fecd63b7d2c4d65d68208dc7bdf57ba%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C638521446390465418%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C0%7C%7C%7C&sdata=pDuv4bnya1eP3Rrs3p4ArNOXw%2FQipWU87q5oZ0VSIe0%3D&reserved=0)
Validate the package for structure before submitting the package for certification to Partner Center - [Run Package Validator tool](https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Flearn.microsoft.com%2Fen-us%2Fconnectors%2Fcustom-connectors%2Fcertification-submission%23step-7-validate-the-package-for-structure&data=05%7C02%7Cshefaali.patankar%40microsoft.com%7C6f0c4adf688f42a7107908dcb6c72a8c%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C638586213737574171%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C0%7C%7C%7C&sdata=9QGRIRe%2FZ%2B1sgUF5JQjY2DeuyxDt8f8lDwpGFEaaBmg%3D&reserved=0)
Initiate a certification request in Partner Center- [Verified publisher certification process](https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Flearn.microsoft.com%2Fen-us%2Fconnectors%2Fcustom-connectors%2Fsubmit-for-certification&data=05%7C02%7Cv-manoharas%40microsoft.com%7C3fecd63b7d2c4d65d68208dc7bdf57ba%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C638521446390471799%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C0%7C%7C%7C&sdata=zbm2px5VZ2ABLBFcORH6tu1fsC9KWdXkP1XJFyluMzw%3D&reserved=0)
Ensure your connector files comply to the Marketplace policies here- [5000 Power Platform Connector Policies for Marketplace](https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Flearn.microsoft.com%2Fen-us%2Flegal%2Fmarketplace%2Fcertification-policies%235000-power-platform-connectors&data=05%7C02%7Cv-manoharas%40microsoft.com%7C3fecd63b7d2c4d65d68208dc7bdf57ba%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C638521446390477736%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C0%7C%7C%7C&sdata=h0pI5J5buo82xxvnBVmrAN8K%2F%2BK8xZlUdcSwzWoqWq0%3D&reserved=0) and also [1000 Marketplace policies](https://learn.microsoft.com/en-us/legal/marketplace/certification-policies#100-general)(as applicable)
Test your connector post certification to provide go-live for deployment- [Testing Guidelines](https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Flearn.microsoft.com%2Fen-us%2Fconnectors%2Fcustom-connectors%2Fcertification-testing&data=05%7C02%7Cv-manoharas%40microsoft.com%7C3fecd63b7d2c4d65d68208dc7bdf57ba%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C638521446390483743%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C0%7C%7C%7C&sdata=Z8pSnhCArFRzEr373o29qM5h5VbV1nlN2gVbv%2Blk1aE%3D&reserved=0)
If you wish to Update your connector in future- [Updating Guidelines](https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Flearn.microsoft.com%2Fen-us%2Fconnectors%2Fcustom-connectors%2Fcertification-updates&data=05%7C02%7Cv-manoharas%40microsoft.com%7C3fecd63b7d2c4d65d68208dc7bdf57ba%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C638521446390489681%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C0%7C%7C%7C&sdata=CvxTWJFE3bTFeJ3nDJoCSpzvimhVV9go4cBG%2FQ4gMvo%3D&reserved=0)
 
For any queries or concerns please get in touch with us on [certificationteam@microsoft.com](mailto:certificationteam@microsoft.com)
 
Many thanks,
Power Platform
Copilot connector & plugin certification team

SagarCIS commented:
Hello @vmanoharas ,

Thank you for approving the PR.

The PR has been merged into the official connector repository, however the newly added actions and triggers (for example Get_InboundTrigger) are not yet visible in Power Automate.

Could you please confirm whether the connector update has been deployed to the production connector registry or if it is still pending deployment?

Best regards,

vmanoharas commented:
Hi @SagarCIS, 

Please follow the steps mentioned above and upload your connector package zip file via Partner Center to certify your connector.
